kernel/posix/pthread_rwlock.c

```diff
@@ -0,0 +1,304 @@
+/*
+ * Copyright (c) 2018 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <kernel.h>
+#include <time.h>
+#include <errno.h>
+#include <sys/types.h>
+
+#define INITIALIZED 1
+#define NOT_INITIALIZED 0
+#define MAX_COUNT 32767
```

Comment:
Why this value? Can't we define this in function of the maximum number of threads in the system?

Reply:
Hi @lpereira Yes its maximum value practically can be equal to number of threads in the system. I will make this modification.